### PR TITLE
AppSync authMode on API.graphql

### PR DESCRIPTION
**Pull request by elorzafe**
AppSync authMode instructions

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: js/api.md [hunk 1]
```diff
@@ -362,6 +362,28 @@ Amplify.configure({
 });
 ```
 
+#### AWS AppSync Multi-Auth
+
+AWS AppSync can support multiple authentication modes on a single API. [LINK TO APPSYNC DOCS](http://docs). In order to use this feature with the Amplify Graphql Client the `API.graphql({...})` function would accept an optional parameter called `authMode`, its value will be one of the supported auth modes:
```
Comment (undefobj) [inline js/api.md]: Multi-auth is for _authorization_ not _authentication_ on a single API. Please update.

File: js/api.md [hunk 2]
```diff
@@ -362,6 +362,28 @@ Amplify.configure({
...
+
+- `API_KEY`
+- `AWS_IAM`
+- `OPENID_CONNECT`
+- `AMAZON_COGNITO_USER_POOLS`
+
+If the authMode is not provided, Amplify will use the default auth mode that was configured:
```
Comment (undefobj) [inline js/api.md]: Where was this configured? In the client constructor? The way this is written it reads as if you're showing how to configure the default mode however the sample code is showing how to override on the method invocation.

**Comment by undefobj** [comment]
Also please update .spelling with `authMode`

File: js/api.md (hunk 1 above)
Comment (elorzafe) [reply js/api.md]: done

**Pull request merged**